packages/integration_test/lib/integration_test_driver_extended.dart

```diff
@@ -127,6 +166,9 @@ Future<void> integrationDriver(
 
   await driver.close();
 
+  if (responseDataCallback != null) {
+    await responseDataCallback(response.data);
+  }
```

Comment:
The regular `integration_test_driver.dart` only does this if all tests passed. Why this difference in behavior?

Reply:
hi, @goderbauer 
usually, the response data is very helpful for us to debug the issue. especially when some tests fail, we need it to help us find the issue.

Reply:
@pdblasi-google Should we keep this logic in sync across both implementations?

Reply:
Yeah, I think it'd be a good idea to keep them consistent. Personally, I'd add a `writeResponseOnFailure` bool to the call (and the one for `integration_test_driver.dart`) that defaults to false to keep the existing behavior, but allow it to be changed when debugging or just if that's the user's preference.

Reply:
updated

Reply:
I don't think the `writeResponseOnFailure` boolean needs to be nullable, but otherwise that change looks good to me.

Reply:
`writeResponseOnFailure` changed to not nullable @pdblasi-google

Reply:
hi @goderbauer  unit test for integration_test_driver_extended.dart added